components/script/dom/bindings/reflector.rs

```diff
@@ -25,20 +25,29 @@ where
     U: DerivedFrom<GlobalScope>,
 {
     let global_scope = global.upcast();
-    unsafe { T::WRAP(GlobalScope::get_cx(), global_scope, None, obj) }
+    unsafe {
+        T::WRAP(
+            GlobalScope::get_cx(),
+            global_scope,
+            None,
+            obj,
+            CanGc::note(),
+        )
+    }
 }
 
 pub fn reflect_dom_object_with_proto<T, U>(
     obj: Box<T>,
     global: &U,
     proto: Option<HandleObject>,
+    can_gc: CanGc,
```

Comment:
is there a future need to pass it here, as opposed to instantiate it inside the function and hide it from the caller?

Reply:
Yes. The purpose is to infect callers so it is clear whether any function can trigger a GC.